lib/detect-testing-library-utils.ts

```diff
@@ -13,48 +18,84 @@ export function detectTestingLibraryUtils<
   TRuleListener extends TSESLint.RuleListener = TSESLint.RuleListener
 >(
   ruleCreate: (
-    context: Readonly<TSESLint.RuleContext<TMessageIds, TOptions>>,
+    context: Readonly<
+      TSESLint.RuleContext<TMessageIds, TOptions> & {
+        settings: TestingLibrarySettings;
+      }
+    >,
     optionsWithDefault: Readonly<TOptions>,
     detectionHelpers: Readonly<DetectionHelpers>
   ) => TRuleListener
 ) {
   return (
-    context: Readonly<TSESLint.RuleContext<TMessageIds, TOptions>>,
+    context: Readonly<
+      TSESLint.RuleContext<TMessageIds, TOptions> & {
+        settings: TestingLibrarySettings;
+      }
+    >,
     optionsWithDefault: Readonly<TOptions>
-  ): TRuleListener => {
-    let isImportingTestingLibrary = false;
+  ): TSESLint.RuleListener => {
+    let isImportingTestingLibraryModule = false;
+    let isImportingCustomModule = false;
 
-    // TODO: init here options based on shared ESLint config
+    // Init options based on shared ESLint settings
+    const customModule = context.settings['testing-library/module'];
 
-    // helpers for Testing Library detection
+    // Helpers for Testing Library detection.
     const helpers: DetectionHelpers = {
-      getIsImportingTestingLibrary() {
-        return isImportingTestingLibrary;
+      getIsTestingLibraryImported() {
+        if (!customModule) {
+          return true;
+        }
+
+        return isImportingTestingLibraryModule || isImportingCustomModule;
+      },
+      canReportErrors() {
+        return this.getIsTestingLibraryImported();
       },
     };
 
-    // instructions for Testing Library detection
+    // Instructions for Testing Library detection.
+    // `ImportDeclaration` must be first in order to know if Testing Library
+    // is imported ASAP.
```

Comment:
I'm still surprised this work. I can elaborate more on this but basically:

1. `detectTestingLibraryUtils` inject a detection rule listener for "ImportDeclaration" which will try to find a Testing Library import
2. before running checks from the original rule listener, `detectTestingLibraryUtils` will make sure Testing Library is imported to prevent original rule listener to be executed

I think this actually works not because the order of the rule listener definitions here, but because imports always happen first in the code linted. I just realized about this rereading my own code and try different orders here and in the concat below. I think I'm gonna remove this comment then to avoid any misunderstanding.